### PR TITLE
Fix some issues flagged by cppcheck

### LINKED ISSUE #1152: va_start() without va_end()
There are some va_start() without a va_end()

https://github.com/GarageGames/Torque3D/blob/4e52824a42108367951c4af77ffd08e979a0b86c/Engine/source/console/console.cpp#L1159

some from:
https://github.com/GarageGames/Torque3D/blob/development/Engine/source/core/strings/stringFunctions.cpp#L387

https://github.com/GarageGames/Torque3D/blob/development/Engine/source/platform/platformAssert.cpp#L176

https://github.com/GarageGames/Torque3D/blob/development/Engine/source/platformMac/macCarbStrings.cpp#L43

https://github.com/GarageGames/Torque3D/blob/development/Engine/source/platformWin32/winConsole.cpp#L137

https://github.com/GarageGames/Torque3D/blob/development/Engine/source/platformX86UNIX/x86UNIXConsole.cpp#L174

found using cppCheck against the engine/source/ folder.

### PR DESCRIPTION
**Pull request by bpay**
Resolves #1152

Any ideas on 3205fea93 ?

### CODE REVIEW

File: Engine/source/core/strings/stringFunctions.cpp [hunk 1]
```diff
@@ -470,7 +472,9 @@ S32 dSscanf(const char *buffer, const char *format, ...)
 #else
    va_list args;
    va_start(args, format);
-   return vsscanf(buffer, format, args);
+   int res = vsscanf(buffer, format, args);
```
Comment (dottools) [inline Engine/source/core/strings/stringFunctions.cpp]: res should be `S32` not `int` for consistency.

File: Engine/source/gfx/bitmap/ddsLoader.cpp [hunk 2]
```diff
@@ -607,7 +607,7 @@ bool DDSFile::read(Stream &s, U32 dropMipCount)
       else if ( mFlags.test( PitchSizeFlag ) )
          mPitchOrLinearSize = getSurfacePitch( dropMipCount );
       else
-         mPitchOrLinearSize = mPitchOrLinearSize; // Do nothing?
```
Comment (dottools) [inline Engine/source/gfx/bitmap/ddsLoader.cpp]: Yeah not sure about this at all.

File: Engine/source/gfx/bitmap/ddsLoader.cpp [hunk 3]
```diff
@@ -607,7 +607,7 @@ bool DDSFile::read(Stream &s, U32 dropMipCount)
...
+         ; // Do nothing?
```
Comment (eightyeight) [inline Engine/source/gfx/bitmap/ddsLoader.cpp]: If it's really a NOOP why not remove the `else` entirely? Also, I can't see what type `mPitchOrLinearSize` is, but maybe check for overloaded assignment?

File: Engine/source/environment/basicClouds.cpp [hunk 4]
```diff
@@ -63,6 +63,12 @@ BasicClouds::BasicClouds()
    mTypeMask |= EnvironmentObjectType | StaticObjectType;
    mNetFlags.set(Ghostable | ScopeAlways);
 
+   mTimeSC =
+   mModelViewProjSC =
+   mTexScaleSC =
+   mTexDirectionSC =
+   mTexOffsetSC = 0;
```
Comment (eightyeight) [inline Engine/source/environment/basicClouds.cpp]: Stylistically I think adding a `0;` on each line is preferable.

File: Engine/source/environment/cloudLayer.cpp [hunk 5]
```diff
@@ -73,15 +73,27 @@ U32 CloudLayer::smVertCount = smVertStride * smVertStride;
...
 CloudLayer::CloudLayer()
-: mBaseColor( 0.9f, 0.9f, 0.9f, 1.0f ),
-  mCoverage( 0.5f ),
+: mLastTime( 0 ),
+  mBaseColor( 0.9f, 0.9f, 0.9f, 1.0f ),
   mExposure( 1.0f ),
-  mWindSpeed( 1.0f ),
-  mLastTime( 0 )
+  mCoverage( 0.5f ),
+  mWindSpeed( 1.0f )
```
Comment (eightyeight) [inline Engine/source/environment/cloudLayer.cpp]: `-Wreorder`?

**Comment by eightyeight** [comment]
Want to stick this in 3.7 if we can.

File: Engine/source/gfx/bitmap/ddsLoader.cpp (hunk 3 above)
Comment (bpay) [inline Engine/source/gfx/bitmap/ddsLoader.cpp]: I left it just as an anchor for the comment, but I guess there's not much point - I can get rid of the else entirely (the var is just a u32). I guess the point of the self-assign was to be able to set a breakpoint there, but if someone really needs that they can always add it back locally.

File: Engine/source/environment/cloudLayer.cpp (hunk 5 above)
Comment (bpay) [inline Engine/source/environment/cloudLayer.cpp]: Yes, figured I would fix the order while I was there. Also makes it easier to see what was left uninintialized. Doing a full reorder pass would be nice, but I am not voluteering for that at the moment. Any takers? :)

Comment (eightyeight) [inline Engine/source/environment/cloudLayer.cpp]: I'll do it after 3.7 :P.

File: Engine/source/gfx/bitmap/ddsLoader.cpp (hunk 3 above)
Comment (eightyeight) [inline Engine/source/gfx/bitmap/ddsLoader.cpp]: :+1:

**Comment by eightyeight** [comment]
Didn't see that commit - thanks!

**Pull request merged**